### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by Abdo9826**
- Copy the files calculate.js and operate.js into a logic/ directory in your project.

- Analyze the files calculate.js and operate.js,

- Import the files in your Calculator component.

- Implement the event handlers you need to use the math logic from calculate.js and operate.js in your React component.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,36 +1,72 @@
...
-          <input
-            type="text"
-            className="calculator-screen z-depth-1"
-            value=""
+          {/* <input
+            type='text'
+            className='calculator-screen z-depth-1'
+            value='12'
             disabled
-          />
```
Comment (KDlamini) [inline src/components/Calculator.js]: Please remove any unused commented code to make your codebase clean and readable. It is a great idea to follow JavaScript's best practices, In your case the [YAGNI principle](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,36 +1,72 @@
...
+            >
               -
             </button>
-            <button type="button" className="operator btn btn-info" value="*">
+            <button
+              type="button"
+              className="operator btn btn-info"
+              name="x"
+              value="x"
+              onClick={this.handleClick}
```
Comment (KDlamini) [inline src/components/Calculator.js:55]: You are missing the `%` button which should handle the `modulus` operation, and the `+-` button to convert numbers to either positive or negative. Kindly add the modulus and +- operators and make sure that they do calculations as expected.

Also please try to follow the layout of the given design.

Your design:
image

Required design:
image

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @Abdo9826 🙋‍♀️,  

Great work so far! 💯 
You have done a great job adding the Math Magicians events for the calculator logic.
There's always room for improvement. There are some issues that you still need to work on improving your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 
✔️  `operate.js` and `calculate.js` components implemented correctly
✔️  State and props are used properly
✔️  Correct PR title and description
✔️  Correct use of Gitflow 
✔️  Commit messages are meaningful
✔️  No linter errors

## Required Changes ♻️

**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 2 above)
Comment (Abdo9826) [reply src/components/Calculator.js:55]: I already git accepted with this design

File: src/components/Calculator.js [hunk 3]
```diff
@@ -1,36 +1,67 @@
...
 
           <div className="calculator-keys">
-            <button type="button" className="operator btn btn-info" value="+">
+            <button
+              type="button"
+              className="operator btn btn-info"
+              name="+"
+              value="+"
+              onClick={this.handleClick}
+            >
```
Comment (omar-muhamad) [inline src/components/Calculator.js:38]: - [ ] As the previous reviewer mentioned You are missing the % button which should handle the modulus operation, and the +- button to convert numbers to either positive or negative. 
So Kindly add the modulus and +- operators and make sure that they do calculations as expected.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Abdo9826,

Nice job making the required changes. Unfortunately, you missed some of them but you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/Calculator.js (hunk 2 above)
Comment (omar-muhamad) [reply src/components/Calculator.js:55]: This requirement didn't fulfilled.

**Review by topeogunleye — APPROVED** [review]
### Hi @Abdo9826, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Gitflow :heavy_check_mark:
 
> ## [Optional] suggestions:
> * Nothing to mention.
> 
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**